numpy/core/src/multiarray/methods.c

```diff
@@ -1008,6 +1008,44 @@ array_getarray(PyArrayObject *self, PyObject *args)
 
 
 static PyObject *
+array_ufunc(PyArrayObject *self, PyObject *args, PyObject *kwds)
+{
+    PyObject *ufunc, *method_name, *normal_args, *ufunc_method, *result;
+
+    if (PyTuple_Size(args) < 2) {
+        PyErr_SetString(PyExc_TypeError,
+                        "__array_ufunc__ requires at least 2 arguments");
+        return NULL;
+    }
+    ufunc = PyTuple_GET_ITEM(args, 0);  /* borrowed reference */
+    if (ufunc == NULL) {
+        return NULL;
+    }
+
+    method_name = PyTuple_GET_ITEM(args, 1);
+    if (method_name == NULL) {
+        return NULL;
+    }
+
+    normal_args = PyTuple_GetSlice(args, 2, PyTuple_GET_SIZE(args));
+    if (normal_args == NULL) {
+        return NULL;
+    }
+
+    ufunc_method = PyObject_GetAttr(ufunc, method_name);
+    if (ufunc_method == NULL) {
+        Py_DECREF(normal_args);
+        return NULL;
+    }
+
+    result = PyObject_Call(ufunc_method, normal_args, kwds);
```

Comment:
@charris - renaming it would beat the whole point of using `super` --the subclass cannot really know which class is above it! See the example above: `MaskedArray` cannot know whether its super will go to `ndarray` or to `Quantity`, and by insisting that the names are the same it doesn't have to check.

(but I agree that it is a bit strange that the base `__array_ufunc__` is not simply part of the `ufunc` machinery -- with its arguments already normalized at the very least it could enter the chain later. But that is for a later PR... For now, I'll just add a `TODO`)